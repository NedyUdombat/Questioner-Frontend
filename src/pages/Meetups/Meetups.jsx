import PropTypes from 'prop-types';
import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Navbar from '../../components/presentational/Navbar/Navbar';
import Image from '../../components/presentational/Image/Image';
import { DEFAULT_IMAGE_URL } from '../../utils/config';
import moment from 'moment';
import { getAllMeetups } from '../../store/modules/meetup';

export class Meetups extends React.Component {
  componentDidMount() {
    this.props.getAllMeetups();
  }

  render() {
    const { meetups } = this.props;
    return (
      <Fragment>
        <Navbar className="nav-dark" />
        <div className="section group">
          {meetups.length === 0 ? (
            <p>No meetups</p>
          ) : (
            meetups.map(meetup => (
              <div className="col span_1_of_4" key={meetup.id}>
                <div className="d-flex justify-content-center">
                  <div className="product-card">
                    <Link to="#">
                      <div className="card-image-top">
                        <Image
                          src={meetup.image || DEFAULT_IMAGE_URL}
                          alt="fr"
                          className="product-image"
                        />
                      </div>
                      <div className="product-card-details">
                        <p className="f-20 ">
                          <b>{meetup.topic}</b>
                        </p>
                        <div className="d-flex justify-content-between mt-1">
                          <small className="f-12">
                            <i>{meetup.organizer_name}</i>
                          </small>
                          <p className="text-blue">
                            {moment(meetup.happening_on).format('DD MMMM YYYY')}
                          </p>
                        </div>
                      </div>
                    </Link>
                  </div>
                </div>
              </div>
            ))
          )}
        </div>
      </Fragment>
    );
  }
}

Meetups.propTypes = {
  getAllMeetups: PropTypes.func.isRequired,
  meetups: PropTypes.array.isRequired,
};

export const mapStateToProps = state => ({
  meetups: state.meetups.meetups,
});

export default connect(
  mapStateToProps,
  { getAllMeetups },
)(Meetups);
