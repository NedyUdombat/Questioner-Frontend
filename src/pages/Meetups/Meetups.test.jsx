import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { Meetups, mapStateToProps } from './Meetups';

let store;

describe('Meetups', () => {
  beforeEach(() => {
    const mockStore = configureMockStore();
    store = mockStore({ meetups: { meetups: [] } });
  });

  it('should render correctly', () => {
    const props = {
      getAllMeetups: jest.fn(),
      meetups: [],
    };
    const component = mount(
      <BrowserRouter>
        <Provider store={store}>
          <Meetups {...props} />
        </Provider>
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with all props', () => {
    const props = {
      getAllMeetups: jest.fn(),
      meetups: [
        {
          id: 'fhwjicgyvufbewf',
          image: 'http://dummyinge.jpg',
          topic: 'topic',
          organizer_name: 'organizer_name',
          happening_on: '2020-05-18T21:28:41+0000',
        },
      ],
    };
    const component = mount(
      <BrowserRouter>
        <Provider store={store}>
          <Meetups {...props} />
        </Provider>
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with all props and without image', () => {
    const props = {
      getAllMeetups: jest.fn(),
      meetups: [
        {
          id: 'fhwjicgyvufbewf',
          topic: 'topic',
          organizer_name: 'organizer_name',
          happening_on: '2020-05-18T21:28:41+0000',
        },
      ],
    };
    const component = mount(
      <BrowserRouter>
        <Provider store={store}>
          <Meetups {...props} />
        </Provider>
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  it('should mount with the correct state', () => {
    const DEFAULT_STATE = {
      meetups: {
        meetups: [],
      },
    };
    const action = mapStateToProps(DEFAULT_STATE);
    expect(action).toEqual(DEFAULT_STATE.meetups);
  });
});
