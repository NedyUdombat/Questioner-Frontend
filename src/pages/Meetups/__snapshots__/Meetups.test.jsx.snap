// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetups should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Meetups
        getAllMeetups={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        meetups={Array []}
      >
        <ConnectFunction
          className="nav-dark"
        >
          <Navbar
            className="nav-dark"
          >
            <div>
              <div
                className="nav nav-dark"
              >
                <input
                  id="nav-check"
                  type="checkbox"
                />
                <div
                  className="nav-header"
                >
                  <Link
                    className="nav-title"
                    to="/"
                  >
                    <a
                      className="nav-title"
                      href="/"
                      onClick={[Function]}
                    >
                      Questioner
                    </a>
                  </Link>
                </div>
                <div
                  className="nav-btn"
                >
                  <label
                    htmlFor="nav-check"
                  >
                    <Icon
                      as="i"
                      color="grey"
                      name="bars"
                      size="big"
                    >
                      <i
                        aria-hidden="true"
                        className="grey bars big icon"
                        onClick={[Function]}
                      />
                    </Icon>
                  </label>
                </div>
                <div
                  className="nav-links"
                >
                  <Link
                    className="btn btn-transparent"
                    to="/meetups"
                  >
                    <a
                      className="btn btn-transparent"
                      href="/meetups"
                      onClick={[Function]}
                    >
                      MEETUPS
                    </a>
                  </Link>
                  <Link
                    className="btn btn-dark-alt"
                    to="/login"
                  >
                    <a
                      className="btn btn-dark-alt"
                      href="/login"
                      onClick={[Function]}
                    >
                      LOGIN
                    </a>
                  </Link>
                  <Link
                    className="btn btn-white-alt-hover"
                    to="/register"
                  >
                    <a
                      className="btn btn-white-alt-hover"
                      href="/register"
                      onClick={[Function]}
                    >
                      REGISTER
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </Navbar>
        </ConnectFunction>
        <div
          className="section group"
        >
          <p>
            No meetups
          </p>
        </div>
      </Meetups>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`Meetups should render correctly with all props 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Meetups
        getAllMeetups={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        meetups={
          Array [
            Object {
              "happening_on": "2020-05-18T21:28:41+0000",
              "id": "fhwjicgyvufbewf",
              "image": "http://dummyinge.jpg",
              "organizer_name": "organizer_name",
              "topic": "topic",
            },
          ]
        }
      >
        <ConnectFunction
          className="nav-dark"
        >
          <Navbar
            className="nav-dark"
          >
            <div>
              <div
                className="nav nav-dark"
              >
                <input
                  id="nav-check"
                  type="checkbox"
                />
                <div
                  className="nav-header"
                >
                  <Link
                    className="nav-title"
                    to="/"
                  >
                    <a
                      className="nav-title"
                      href="/"
                      onClick={[Function]}
                    >
                      Questioner
                    </a>
                  </Link>
                </div>
                <div
                  className="nav-btn"
                >
                  <label
                    htmlFor="nav-check"
                  >
                    <Icon
                      as="i"
                      color="grey"
                      name="bars"
                      size="big"
                    >
                      <i
                        aria-hidden="true"
                        className="grey bars big icon"
                        onClick={[Function]}
                      />
                    </Icon>
                  </label>
                </div>
                <div
                  className="nav-links"
                >
                  <Link
                    className="btn btn-transparent"
                    to="/meetups"
                  >
                    <a
                      className="btn btn-transparent"
                      href="/meetups"
                      onClick={[Function]}
                    >
                      MEETUPS
                    </a>
                  </Link>
                  <Link
                    className="btn btn-dark-alt"
                    to="/login"
                  >
                    <a
                      className="btn btn-dark-alt"
                      href="/login"
                      onClick={[Function]}
                    >
                      LOGIN
                    </a>
                  </Link>
                  <Link
                    className="btn btn-white-alt-hover"
                    to="/register"
                  >
                    <a
                      className="btn btn-white-alt-hover"
                      href="/register"
                      onClick={[Function]}
                    >
                      REGISTER
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </Navbar>
        </ConnectFunction>
        <div
          className="section group"
        >
          <div
            className="col span_1_of_4"
            key="fhwjicgyvufbewf"
          >
            <div
              className="d-flex justify-content-center"
            >
              <div
                className="product-card"
              >
                <Link
                  to="/meetups/fhwjicgyvufbewf"
                >
                  <a
                    href="/meetups/fhwjicgyvufbewf"
                    onClick={[Function]}
                  >
                    <div
                      className="card-image-top"
                    >
                      <Image
                        alt="fr"
                        className="product-image"
                        src="http://dummyinge.jpg"
                      >
                        <img
                          alt="fr"
                          className="product-image"
                          src="http://dummyinge.jpg"
                        />
                      </Image>
                    </div>
                    <div
                      className="product-card-details"
                    >
                      <p
                        className="f-20 "
                      >
                        <b>
                          topic
                        </b>
                      </p>
                      <div
                        className="d-flex justify-content-between mt-1"
                      >
                        <small
                          className="f-12"
                        >
                          <i>
                            organizer_name
                          </i>
                        </small>
                        <p
                          className="text-blue"
                        >
                          18 May 2020
                        </p>
                      </div>
                    </div>
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </Meetups>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`Meetups should render correctly with all props and without image 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Meetups
        getAllMeetups={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        meetups={
          Array [
            Object {
              "happening_on": "2020-05-18T21:28:41+0000",
              "id": "fhwjicgyvufbewf",
              "organizer_name": "organizer_name",
              "topic": "topic",
            },
          ]
        }
      >
        <ConnectFunction
          className="nav-dark"
        >
          <Navbar
            className="nav-dark"
          >
            <div>
              <div
                className="nav nav-dark"
              >
                <input
                  id="nav-check"
                  type="checkbox"
                />
                <div
                  className="nav-header"
                >
                  <Link
                    className="nav-title"
                    to="/"
                  >
                    <a
                      className="nav-title"
                      href="/"
                      onClick={[Function]}
                    >
                      Questioner
                    </a>
                  </Link>
                </div>
                <div
                  className="nav-btn"
                >
                  <label
                    htmlFor="nav-check"
                  >
                    <Icon
                      as="i"
                      color="grey"
                      name="bars"
                      size="big"
                    >
                      <i
                        aria-hidden="true"
                        className="grey bars big icon"
                        onClick={[Function]}
                      />
                    </Icon>
                  </label>
                </div>
                <div
                  className="nav-links"
                >
                  <Link
                    className="btn btn-transparent"
                    to="/meetups"
                  >
                    <a
                      className="btn btn-transparent"
                      href="/meetups"
                      onClick={[Function]}
                    >
                      MEETUPS
                    </a>
                  </Link>
                  <Link
                    className="btn btn-dark-alt"
                    to="/login"
                  >
                    <a
                      className="btn btn-dark-alt"
                      href="/login"
                      onClick={[Function]}
                    >
                      LOGIN
                    </a>
                  </Link>
                  <Link
                    className="btn btn-white-alt-hover"
                    to="/register"
                  >
                    <a
                      className="btn btn-white-alt-hover"
                      href="/register"
                      onClick={[Function]}
                    >
                      REGISTER
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </Navbar>
        </ConnectFunction>
        <div
          className="section group"
        >
          <div
            className="col span_1_of_4"
            key="fhwjicgyvufbewf"
          >
            <div
              className="d-flex justify-content-center"
            >
              <div
                className="product-card"
              >
                <Link
                  to="/meetups/fhwjicgyvufbewf"
                >
                  <a
                    href="/meetups/fhwjicgyvufbewf"
                    onClick={[Function]}
                  >
                    <div
                      className="card-image-top"
                    >
                      <Image
                        alt="fr"
                        className="product-image"
                        src="https://react.semantic-ui.com/images/wireframe/image.png"
                      >
                        <img
                          alt="fr"
                          className="product-image"
                          src="https://react.semantic-ui.com/images/wireframe/image.png"
                        />
                      </Image>
                    </div>
                    <div
                      className="product-card-details"
                    >
                      <p
                        className="f-20 "
                      >
                        <b>
                          topic
                        </b>
                      </p>
                      <div
                        className="d-flex justify-content-between mt-1"
                      >
                        <small
                          className="f-12"
                        >
                          <i>
                            organizer_name
                          </i>
                        </small>
                        <p
                          className="text-blue"
                        >
                          18 May 2020
                        </p>
                      </div>
                    </div>
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </Meetups>
    </Provider>
  </Router>
</BrowserRouter>
`;
